Gems/Atom/Tools/MaterialCanvas/Assets/MaterialCanvas/TestData/abs.materialcanvasnode.azasset

```diff
@@ -0,0 +1,57 @@
+{
+    "Type": "JsonSerialization",
+    "Version": 1,
+    "ClassName": "DynamicNodeConfig",
+    "ClassData": {
+        "category": "Math Functions",
+        "title": "abs",
+        "inputSlots": [
+            {
+                "name": "inValue",
+                "displayName": "Value",
+                "description": "Value",
+                "supportedDataTypes": [
+                    "float",
```

Comment:
I'm wondering if we need a different way to describe the type support. Once you add the int types, 16 bit types, packed 8-bit unorm/snorm types, matrices, etc., it will be quite exhausting to list all of them for each operation that supports a collection of them.

Reply:
Yes, as mentioned above, was thinking the same thing. Will need a more generic and flexible way to describe types & compatibility but it will have to be compatible with what's supported by the graph model. The specific types are also used to determine which control shows up for properties on a node. Once we get the native AZStd::any support in the serializer, there might be a way to data drive and automatically register all of the supported types, compatibility, and conversion. Like you mentioned, maybe we just list vector or scalar in the supported types field and convert that to more explicit types and combinations after it loads the configurations in code.